app/controllers/projects/project_follows_controller.rb

```diff
@@ -1,25 +1,19 @@
 class Projects::ProjectFollowsController < ApplicationController
-  before_action :get_project
-  before_action :get_episode
+  load_and_authorize_resource :project, find_by: :url
+
+  def index
+    @users = @project.project_followers
+  end
 
   def create
-    return redirect_to project_path(@episode, @project) if current_user.project_followings.include?(@project)
-    current_user.project_followings << @project
-    redirect_to project_path(@episode, @project), notice: "You are now watching #{@project.title}"
+    current_user.project_followings= current_user.project_followings | [@project]
```

Comment:
This is really saving the content that you are assigning?

Reply:
```ruby
irb(main):002:0> ['a', 'b'] | ['a', 'c']
=> ["a", "b", "c"]
```

Reply:
@hennevogel, I know what `|` is doing, but are you saving the data? I mean, I don't see any `current_user.save`

Reply:
[CollectionProxy](http://api.rubyonrails.org/classes/ActiveRecord/Associations/CollectionProxy.html) assignment is direct, unlike attribute assignment...

```
[1] pry(main)> User.last
  User Load (0.8ms)  SELECT  `users`.* FROM `users`  ORDER BY `users`.`id` DESC LIMIT 1
rails_console: No such file or directory - which
=> #<User id: 2, uid: nil, name: "heino", email: "heino@heino.de", created_at: "2018-07-09 14:30:21", updated_at: "2018-07-09 14:30:21", location: nil>
[2] pry(main)> User.last.project_followings
  User Load (1.0ms)  SELECT  `users`.* FROM `users`  ORDER BY `users`.`id` DESC LIMIT 1
=>   Project Load (1.1ms)  SELECT `projects`.* FROM `projects` INNER JOIN `project_follows` ON `projects`.`id` = `project_follows`.`project_id` WHERE `project_follows`.`user_id` = 2
[#<Project:0x0000000007449308
  id: 2,
  title: "archived",
  description: "## schwengel :uganda: ",
  created_at: Fri, 06 Jul 2018 23:16:22 UTC +00:00,
  updated_at: Tue, 10 Jul 2018 00:02:01 UTC +00:00,
  originator_id: 1,
  likes_count: 1,
  memberships_count: 1,
  aasm_state: "project",
  avatar_file_name: nil,
  avatar_content_type: nil,
  avatar_file_size: nil,
  avatar_updated_at: nil,
  url: "hosenscheisser">]
[3] pry(main)> User.last.project_followings = User.last.project_followings | [Project.find(1)]
  User Load (1.0ms)  SELECT  `users`.* FROM `users`  ORDER BY `users`.`id` DESC LIMIT 1
  User Load (0.8ms)  SELECT  `users`.* FROM `users`  ORDER BY `users`.`id` DESC LIMIT 1
  Project Load (1.0ms)  SELECT  `projects`.* FROM `projects` WHERE `projects`.`id` = 1 LIMIT 1
  Project Load (1.2ms)  SELECT `projects`.* FROM `projects` INNER JOIN `project_follows` ON `projects`.`id` = `project_follows`.`project_id` WHERE `project_follows`.`user_id` = 2
  Project Load (1.2ms)  SELECT `projects`.* FROM `projects` INNER JOIN `project_follows` ON `projects`.`id` = `project_follows`.`project_id` WHERE `project_follows`.`user_id` = 2
   (0.5ms)  BEGIN
  SQL (0.8ms)  INSERT INTO `project_follows` (`user_id`, `project_id`, `created_at`, `updated_at`) VALUES (2, 1, '2018-07-10 12:43:10', '2018-07-10 12:43:10')
   (6.7ms)  COMMIT
=> [#<Project:0x0000000007517d20
  id: 2,
  title: "archived",
  description: "## schwengel :uganda: ",
  created_at: Fri, 06 Jul 2018 23:16:22 UTC +00:00,
  updated_at: Tue, 10 Jul 2018 00:02:01 UTC +00:00,
  originator_id: 1,
  likes_count: 1,
  memberships_count: 1,
  aasm_state: "project",
  avatar_file_name: nil,
  avatar_content_type: nil,
  avatar_file_size: nil,
  avatar_updated_at: nil,
  url: "hosenscheisser">,
 #<Project:0x0000000004f2ac98
  id: 1,
  title: "zwei",
  description: "entenstuben",
  created_at: Fri, 06 Jul 2018 23:15:49 UTC +00:00,
  updated_at: Mon, 09 Jul 2018 14:35:12 UTC +00:00,
  originator_id: 1,
  likes_count: nil,
  memberships_count: 1,
  aasm_state: "project",
  avatar_file_name: nil,
  avatar_content_type: nil,
  avatar_file_size: nil,
  avatar_updated_at: nil,
  url: "schlawiner">]
[4] pry(main)> User.last.reload.project_followings
  User Load (1.0ms)  SELECT  `users`.* FROM `users`  ORDER BY `users`.`id` DESC LIMIT 1
  User Load (1.1ms)  SELECT  `users`.* FROM `users` WHERE `users`.`id` = 2 LIMIT 1
=>   Project Load (1.2ms)  SELECT `projects`.* FROM `projects` INNER JOIN `project_follows` ON `projects`.`id` = `project_follows`.`project_id` WHERE `project_follows`.`user_id` = 2
[#<Project:0x00000000076d6bc0
  id: 1,
  title: "zwei",
  description: "entenstuben",
  created_at: Fri, 06 Jul 2018 23:15:49 UTC +00:00,
  updated_at: Mon, 09 Jul 2018 14:35:12 UTC +00:00,
  originator_id: 1,
  likes_count: nil,
  memberships_count: 1,
  aasm_state: "project",
  avatar_file_name: nil,
  avatar_content_type: nil,
  avatar_file_size: nil,
  avatar_updated_at: nil,
  url: "schlawiner">,
 #<Project:0x00000000076d65f8
  id: 2,
  title: "archived",
  description: "## schwengel :uganda: ",
  created_at: Fri, 06 Jul 2018 23:16:22 UTC +00:00,
  updated_at: Tue, 10 Jul 2018 00:02:01 UTC +00:00,
  originator_id: 1,
  likes_count: 1,
  memberships_count: 1,
  aasm_state: "project",
  avatar_file_name: nil,
  avatar_content_type: nil,
  avatar_file_size: nil,
  avatar_updated_at: nil,
  url: "hosenscheisser">]
```